be/op.h

```diff
@@ -169,12 +232,37 @@ M0_INTERNAL int m0_be_op_tick_ret(struct m0_be_op *op,
 				  struct m0_fom   *fom,
 				  int              next_state);
 
+
+M0_INTERNAL void m0_be_op_make_set_and(struct m0_be_op *op);
+M0_INTERNAL void m0_be_op_make_set_or(struct m0_be_op *op);
+
 /**
- * Adds @child to @parent, making the latter an "op set".
+ * Adds child to parent.
+ *
+ * @note parent MUST be an op set already.
+ * @see m0_be_op_make_set_and(), m0_be_op_make_set_or().
  */
 M0_INTERNAL void m0_be_op_set_add(struct m0_be_op *parent,
 				  struct m0_be_op *child);
 
+
+/**
+ * This function MUST be called after addition all ops to the op set, but before
+ * calling m0_be_op_wait() or m0_be_op_tick_ret().
+ *
+ * @note other ops MUST NOT be added be added to the op set after this call.
+ * @note the op set MUST have at least one children.
+ * @note op set children SHOULD NOT be finalized before this function completes.
+ */
+M0_INTERNAL void m0_be_op_set_add_finish(struct m0_be_op *op);
+
+/**
+ * Returns the op that directly lead to this op set to become done.
```

Comment:
\*suggestion\* "Returns the op that directly le~~a~~d ~~to~~ this op set to M0_BOS_DONE state"; or we could just simplify it: "Returns the op that caused the op set to become done".